exporter/awsprometheusremotewriteexporter/auth.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsprometheusremotewriteexporter
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+// signingRoundTripper is a Custom RoundTripper that performs AWS Sig V4
+type signingRoundTripper struct {
+	transport http.RoundTripper
+	signer    *v4.Signer
+	cfg       *aws.Config
+	service   string
+}
+
+// RoundTrip signs each outgoing request
+func (si *signingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	reqBody, err := req.GetBody()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the body
+	content, err := ioutil.ReadAll(reqBody)
```

Comment:
Would it make a sense to put a LimitReader here?

```
ioutil.ReadAll(io.LimitReader{
   R: reqBody,
   N: some large enough number but not too large
})
```


If there is a malicious body, it can cause out-of-memory. Not super critical for secure connections but a thought.

Reply:
hmm sounds like a good idea but i'm not sure what threshold would suffice